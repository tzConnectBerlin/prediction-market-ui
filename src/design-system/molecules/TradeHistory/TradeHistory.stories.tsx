import { Story, Meta } from '@storybook/react/types-6-0';
import { GridColDef } from '@material-ui/data-grid';
import { TradeHistory, TradeHistoryProps } from './TradeHistory';

export default {
  title: 'Molecule/TradeHistory',
  component: TradeHistory,
} as Meta;

const columnList: GridColDef[] = [
  { field: 'date', headerName: 'Date', type: 'date', flex: 1 },
  { field: 'address', headerName: 'User Address', flex: 1.5 },
  { field: 'outcome', headerName: 'Outcome', flex: 1 },
  {
    field: 'quantity',
    headerName: 'Quantity',
    type: 'number',
    flex: 0.8,
  },
];

const rowList = [
  {
    id: 1,
    date: '5 MAR 2021',
    address: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'no',
    quantity: 200,
  },
  {
    id: 2,
    date: '4 MAR 2021',
    address: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'yes',
    quantity: 20,
  },
  {
    id: 3,
    date: '3 MAR 2021',
    address: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'no',
    quantity: 50,
  },
  {
    id: 4,
    date: '2 MAR 2021',
    address: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'yes',
    quantity: 100,
  },
];
const Template: Story<TradeHistoryProps> = (args) => <TradeHistory {...args} />;

export const Default = Template.bind({});
Default.args = {
  rows: rowList,
  columns: columnList,
  hideFooterPagination: true,
};

export const Pageination = Template.bind({});
Pageination.args = {
  rows: rowList,
  columns: columnList,
  pageSize: 5,
};
