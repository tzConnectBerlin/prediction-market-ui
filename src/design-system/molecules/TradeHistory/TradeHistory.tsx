import React from 'react';
import { DataGrid, DataGridProps } from '@material-ui/data-grid';
import { Link, Paper, Typography } from '@material-ui/core';
import styled from '@emotion/styled';

const StyledWrapper = styled(Paper)`
  padding: 1rem;
`;

const StyledLink = styled.div`
  text-align: center;
  padding-top: 1rem;
`;

export interface TradeHistoryProps extends DataGridProps {
  title?: string;
  linkText?: string;
  linkAddress: string;
}

export const TradeHistory: React.FC<TradeHistoryProps> = ({
  title = 'Trade History',
  linkAddress,
  linkText = 'See All',
  ...rest
}) => {
  return (
    <StyledWrapper square>
      <Typography variant="h5" component="h1">
        {title}
      </Typography>
      <DataGrid {...rest} autoHeight />
      <StyledLink>
        <Link href={linkAddress} variant="body2" underline="none">
          {linkText}
        </Link>
      </StyledLink>
    </StyledWrapper>
  );
};
