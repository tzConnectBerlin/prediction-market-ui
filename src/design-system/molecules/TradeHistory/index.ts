export { TradeHistory as default, TradeHistory } from './TradeHistory';
export type { TradeHistoryProps } from './TradeHistory';
