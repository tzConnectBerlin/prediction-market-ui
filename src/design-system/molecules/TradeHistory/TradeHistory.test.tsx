import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { GridColDef } from '@material-ui/data-grid';
import { TradeHistory } from './TradeHistory';

const columnList: GridColDef[] = [
  { field: 'date', headerName: 'Date', type: 'date', flex: 1 },
  { field: 'address', headerName: 'User Address', flex: 1 },
  { field: 'outcome', headerName: 'Outcome', flex: 1 },
  {
    field: 'quantity',
    headerName: 'Quantity',
    type: 'number',
    flex: 1,
  },
];

const rowList = [
  {
    id: 1,
    date: '5 MAR 2021',
    address: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'no',
    quantity: 200,
  },
  {
    id: 2,
    date: '4 MAR 2021',
    address: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'yes',
    quantity: 20,
  },
  {
    id: 3,
    date: '3 MAR 2021',
    address: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'no',
    quantity: 50,
  },
  {
    id: 4,
    date: '2 MAR 2021',
    address: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'yes',
    quantity: 100,
  },
];

describe('Snapshot testing Table Component', () => {
  it('renders correctly', () => {
    const Table = renderer.create(<TradeHistory columns={columnList} rows={rowList} />).toJSON();
    expect(Table).toMatchSnapshot();
  });
});

describe('Element testing Table Component', () => {
  it('render correctly with mock data', async () => {
    const { getByText } = render(
      <TradeHistory columns={columnList} rows={rowList} title="Test Title" />,
    );

    expect(getByText(/Test Title/i)).toBeInTheDocument();
    expect(getByText(/5 MAR 2021/i)).toBeInTheDocument();
  });
});
