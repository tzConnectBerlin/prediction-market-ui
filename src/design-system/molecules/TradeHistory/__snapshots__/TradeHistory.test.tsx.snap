// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Table Component renders correctly 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  padding: 1rem;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
}

.emotion-2 {
  text-align: center;
  padding-top: 1rem;
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.8125rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 4px 5px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #3f51b5;
  border-radius: 0.2em;
  padding: 0.2em 1.2em;
  border: solid 2px transparent;
}

.emotion-3::-moz-focus-inner {
  border-style: none;
}

.emotion-3.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-3 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-3:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(63, 81, 181, 0.04);
}

@media (hover: none) {
  .emotion-3:hover {
    background-color: transparent;
  }
}

.emotion-3.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-3:hover {
  border-width: 2px!important;
}

.emotion-3:disabled {
  border-color: transparent;
}

.emotion-4 {
  width: 100%;
  display: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
  -webkit-box-pack: inherit;
  -ms-flex-pack: inherit;
  -webkit-justify-content: inherit;
  justify-content: inherit;
}

.emotion-5 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation1 emotion-0"
>
  <h1
    className="MuiTypography-root MuiTypography-h5 emotion-1"
  >
    Trade History
  </h1>
  <div
    aria-colcount={4}
    aria-label="grid"
    aria-multiselectable={false}
    aria-rowcount={4}
    className="MuiDataGrid-root MuiDataGrid-root MuiDataGrid-autoHeight"
    role="grid"
  >
    <div
      className="MuiDataGrid-main"
    >
      <div
        className="MuiDataGrid-overlay"
        error={[TypeError: Cannot read property 'parentElement' of null]}
        errorInfo={
          Object {
            "componentStack": "
    at /Users/shohreh/Projects/prediction-market-ui/node_modules/@material-ui/data-grid/dist/index-cjs.js:15:68708
    at div
    at is (/Users/shohreh/Projects/prediction-market-ui/node_modules/@material-ui/data-grid/dist/index-cjs.js:15:157289)
    at as (/Users/shohreh/Projects/prediction-market-ui/node_modules/@material-ui/data-grid/dist/index-cjs.js:15:156723)
    at div
    at /Users/shohreh/Projects/prediction-market-ui/node_modules/@material-ui/data-grid/dist/index-cjs.js:15:32439
    at NoSsr (/Users/shohreh/Projects/prediction-market-ui/node_modules/@material-ui/core/node/NoSsr/NoSsr.js:33:5)
    at /Users/shohreh/Projects/prediction-market-ui/node_modules/@material-ui/data-grid/dist/index-cjs.js:15:157982
    at /Users/shohreh/Projects/prediction-market-ui/node_modules/@material-ui/data-grid/dist/index-cjs.js:15:166127
    at div
    at MuiPaper-root (/Users/shohreh/Projects/prediction-market-ui/node_modules/@emotion/react/dist/emotion-element-cb6e9ca7.cjs.dev.js:35:23)
    at Paper (/Users/shohreh/Projects/prediction-market-ui/node_modules/@material-ui/core/node/Paper/Paper.js:91:44)
    at Styled(Component) (/Users/shohreh/Projects/prediction-market-ui/node_modules/@emotion/react/dist/emotion-element-cb6e9ca7.cjs.dev.js:35:23)
    at TradeHistory (/Users/shohreh/Projects/prediction-market-ui/src/design-system/molecules/TradeHistory/TradeHistory.tsx:23:3)",
          }
        }
        style={
          Object {
            "top": 56,
          }
        }
      >
        An error occurred.
      </div>
    </div>
  </div>
  <div
    className="emotion-2"
  >
    <button
      bordercolor="transparent"
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root emotion-3"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label emotion-4"
      >
        See All
      </span>
      <span
        className="MuiTouchRipple-root emotion-5"
      />
    </button>
  </div>
</div>
`;
